pkg/apis/serving/k8s_validation.go

```diff
@@ -451,6 +452,10 @@ func validateReadinessProbe(p *corev1.Probe) *apis.FieldError {
 		errs = errs.Also(apis.ErrOutOfBoundsValue(p.PeriodSeconds, 0, math.MaxInt32, "periodSeconds"))
 	}
 
+	if p.InitialDelaySeconds < 0 {
+		errs = errs.Also(apis.ErrInvalidValue(strconv.Itoa(int(p.InitialDelaySeconds))+": must be greater than or equal to 0", "initialDelaySeconds"))
```

Comment:
Would `ErrOutOfBoundsValue` be a more appropriate error?
See below and above checks.